definitions/checks/candlepin/db_index.rb

```diff
@@ -0,0 +1,21 @@
+module Checks
+  module Candlepin
+    class DBIndex < ForemanMaintain::Check
+      metadata do
+        description 'Make sure Candlepin DB indexes are OK'
+        label :candlepin_db_index
+        tags :db_index
+        for_feature :candlepin_database
+        confine do
+          feature(:candlepin_database)&.local?
+        end
+      end
+
+      def run
+        status, output = feature(:candlepin_database).amcheck
+
+        assert(status == 0 || status.nil?, "Candlepin DB indexes have issues:\n#{output}")
```

Comment:
`status` is `nil` when `db.amcheck` returned early, without actually running amcheck. I considered returning `[0, '']`, but this felt wrong, as the status is really not `0` (`OK`).
I could also write this as
```suggestion
        assert(status == 0 , "Candlepin DB indexes have issues:\n#{output}") unless status.nil?
```

what do you prefer?

Reply:
We could also do smth like
```
        unless status.nil?
          assert(status == 0, "Candlepin DB indexes have issues:\n#{output}")
        else
          skip "amcheck not possible"
        end
```